contrib/net/http/make_responsewriter.go

```diff
@@ -81,6 +81,6 @@ func wrapResponseWriter(w http.ResponseWriter, span ddtrace.Span) http.ResponseW
 		w = mw
 	}
 
-	return w
+	return w, mw
```

Comment:
Why not simply change the type of the return value? It will still work, because it still implements `http.ResponseWriter`. This way you're just returning the same thing twice under different types.

Reply:
Unfortunately, `w` is a `http.ResponseWriter` since we're inventing them on the fly by combining things into unnamed struct types. At the end, `w` is *not* a `*responseWriter`, but a struct containing one, so we have to return it independently. I agree it's kind of ugly, but I'm not sure a better way without more major refactoring.